### ISSUE TITLE
[Cross-Case Skill Discovery] VitsModel (train mode) case

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#61]
# [Cross-Case Skill Discovery] VitsModel (train mode) case

**Experiment:** [`2026-04-cross-case-skill-discovery`](https://github.com/penguinwu/oss-model-graph-break-corpus/tree/main/discovery/experiments/2026-04-cross-case-skill-discovery)
**Master plan:** [`plan.md`](https://github.com/penguinwu/oss-model-graph-break-corpus/blob/main/discovery/experiments/2026-04-cross-case-skill-discovery/plan.md) — methodology, matrix, questions, what we record. Read this before launching.
**Analysis methodology:** [`per-case-analysis SKILL`](https://github.com/penguinwu/oss-model-graph-break-corpus/blob/main/discovery/skills/per-case-analysis/SKILL.md) — 7-phase blueprint (Phase 0 audit + A-F). Follow this when the run completes.
**Umbrella:** #60
**Owner:** Otter
**Status:** Ready (case file authored + smoke-tested; pre-launch checklist in progress)

---

This issue holds the **model-specific** setup, results, and discussion for case `vits_model_train` (`VitsModel (train mode)`). The methodology is in the master plan — do not restate it here.

## Case-specific setup (fill in when case is authored)

| Item | Value |
|---|---|
| Case ID | `vits_model_train` |
| Model | `VitsModel (train mode)` |
| Case file | `discovery/cases/vits_model_train.py` |
| Baseline (perf) | `discovery/cases/vits_model_train.baseline.json` |
| Pristine source backups | `/tmp/discovery-runs/vits_model_train/*.original` |
| Baseline eager output | `/tmp/discovery-runs/vits_model_train/baseline_eager_output.pt` |
| Validator | `/tmp/discovery-runs/vits_model_train/validate.py` |
| Source files agent may edit | modeling_vits.py + the baseline script |
| Source files OFF-LIMITS | sdpa_attention.py, decomposition tables, anything else |

## Baseline numbers (no fix, with breaks)

- `graph_break_count`: 22
- `eager_ms` (tier-1): 95.77
- `compiled_ms` (tier-1): 86.09
- `speedup`: 1.11x
- `compile_s`: 58.81
- `max_diff_compiled_vs_eager`: deterministic; correctness uses prefix-clamp due to nondet waveform length

**Notes:** Train mode (model.train() — required to surface layerdrop + stochastic-dropout breaks). Single watched source file. Correctness compares min(eager_len, compiled_len) prefix.

### Canonical (snapshotted from master plan on 2026-04-25)

- [ ] *Case file imports cleanly:* `python -c "from discovery.cases.<case_id> import get_case_spec; print(get_case_spec().case_id)"` returns the case_id without exception.
- [ ] *Baseline.json present:* `ls discovery/cases/<case_id>.baseline.json` exists with reasonable speedup numbers.
- [ ] *Pristine source backups:* `ls /tmp/discovery-runs/<case_id>/*.original` returns expected files (one `.original` per watched source file).
- [ ] *Baseline eager output saved:* `ls /tmp/discovery-runs/<case_id>/baseline_eager_output.pt` exists.
- [ ] *Validate runs cleanly:* `python /tmp/discovery-runs/<case_id>/validate.py` reports `import_ok`, `eager_ok`, `compile_ok` all true; `graph_break_count` matches the case's documented baseline (in baseline.json or case docstring).
- [ ] *Backend confirmed:* `baseline_*.py` and `validate.py` use `torch.compile(model)` with default backend (inductor) — NOT `backend="eager"`. Discovery uses inductor; sweep uses eager (per design.md §2.1).
- [ ] *Watched files set:* case spec lists exactly the files agent may edit; no shared infra (`sdpa_attention.py`, decomposition tables) included; allowed list matches what the prompt declares.
- [ ] *Skill file accessible:* `ls discovery/skills/debug-graph-breaks/SKILL.md` exists (used in the with-skill arm).
- [ ] *No conflicting trials in flight:* `ps -ef | grep "discovery.run_case --case <case_id>"` returns nothing — sequential trials would conflict on shared model files.

If any item is flagged, fix or document the deviation BEFORE launching. Don't launch with unchecked items.

### Case-specific additions (VitsModel)

- [ ] *Train mode confirmed:* case file calls `model.train()` not `.eval()` — required to surface layerdrop + stochastic-dropout breaks specific to train mode.
- [ ] *Stochastic duration handling:* baseline_vits.py + validate.py use `min(eager_len, compiled_len)` prefix-clamp because VitsStochasticDurationPredictor produces nondeterministic waveform lengths between eager and compiled runs.
- [ ] *Single source file:* watched_files = [modeling_vits.py, baseline_vits.py] only — no separate vision encoder source like Mistral3's pixtral.

If any item above is unchecked, fix or document the deviation BEFORE launching.

## Launch command (when ready)

```bash
cd ~/projects/oss-model-graph-break-corpus
nohup /home/pengwu/envs/torch211/bin/python -m discovery.run_case \
  --case vits_model_train \
  --variants V0,V2,V4,V6 \
  --skills none,/path/to/debug-graph-breaks/SKILL.md \
  --n 3 \
  --timeout 1800 \
  > /tmp/discovery-runs/vits_model_train/launches/launch.log 2>&1 &
```

(Drop the second `--skills` arm to `none` only if running without the skill axis.)

## Trials

(Filled live as trials complete. One row per trial.)

| Trial | exit | elapsed | gb_final | max_diff | speedup | strategy summary |
|---|---|---|---|---|---|---|
| — | — | — | — | — | — | — |

## Post-run checklist

Tick after the harness completes and before declaring the case done.

- [ ] *Phase 0 audit clean* (per `per-case-analysis SKILL`): artifact completeness, .original SHA-verified against pristine source, no serious flags, internal consistency, value bounds sane, stream integrity OK.
- [ ] *validation_v2 added* to all trial result.json files (run `python -m discovery.revalidate --case vits_model_train --run-id <run_id>` if not native to the per-case validate.py).
- [ ] *fingerprints.csv produced* per Phase A, classifying every trial.
- [ ] *Phase B aggregations* computed (fix_status × variant × skill table; perf distributions; strategy clusters).
- [ ] *Phase C-D-E walked* in the findings doc — questions answered, surprises documented, open observations flagged honestly (no invented mechanisms).
- [ ] *Findings doc landed* at `discovery/experiments/2026-04-cross-case-skill-discovery/reports/vits_model_train/findings.md` per Phase F.
- [ ] *PR opened* adding the report; PR description links back to this issue.

## Final report (deliverable)

PR adds `discovery/experiments/2026-04-cross-case-skill-discovery/reports/vits_model_train/findings.md` + `fingerprints.csv` per the master plan's report convention. PR description links back to this issue. Merge → this issue moves to Done.

## Lessons learned (harvested at case wrap)

(Filled in when the case wraps. Anything this case taught us about the harness, the variant catalog, the fingerprint axes, the master plan, the methodology — short bullets. Cross-case synthesis pulls from these.)

## Status updates (comment thread)

Comments below should mark per-trial milestones, surprises, and pivots specific to this case. Cross-case observations belong on the umbrella (#60).

**Comment by penguinwu** [comment]
**Closing — VITS case complete.**

Field report shipped (`7ae802b`) with all 15 corrected-validator trials in:
- 12 originally clean + 3 parallel re-run replacements (after the [filesystem-contamination 3-layer protection](https://github.com/penguinwu/oss-model-graph-break-corpus/blob/main/discovery/EXPERIMENT_LIFECYCLE.md#built-in-contamination-protections) shipped earlier in the day caught + fixed an isolation bug)

Findings live at: [`discovery/experiments/2026-04-cross-case-skill-discovery/reports/vits_model_train/findings.md`](https://github.com/penguinwu/oss-model-graph-break-corpus/blob/main/discovery/experiments/2026-04-cross-case-skill-discovery/reports/vits_model_train/findings.md)

Format is field-report (per Peng): no statistical generalization (N=1 model), focus on (1) methodology — how a real fix is verified, (2) the 5 break shapes attacked, (3) working strategies per break shape, (4) shortcut strategies per break shape, (5) agent behaviors observed, (6) the 4 project questions answered as observations from this experiment, (7) caveats + open questions.

Open follow-ups (tracked as separate items, not blocking this issue):
- S7 → debug-graph-breaks skill PR (to Arsh's fork)
- Apparent-fix vs correct-fix audit (max_diff floor)
- S7 generalization probe on non-TTS data-dep arange
- Negative-discovery extraction tool from stream.jsonl

Report is ready for review.

**Comment by penguinwu** [comment]
**📦→🔬 Repo migration redirect (2026-05-02)**

This issue's referenced artifacts have moved to **[penguinwu/pt2-skill-discovery](https://github.com/penguinwu/pt2-skill-discovery)** as part of the corpus → pt2-skill-discovery split (corpus commit `06720e9`).

Path translation: `discovery/X` in this issue's body now lives at `X` in pt2-skill-discovery (drop the `discovery/` prefix). Specifically:
- `discovery/experiments/<slug>/...` → [`experiments/<slug>/...`](https://github.com/penguinwu/pt2-skill-discovery/tree/main/experiments)
- `discovery/cases/...` → [`cases/...`](https://github.com/penguinwu/pt2-skill-discovery/tree/main/cases)
- `discovery/skills/...` → [`skills/...`](https://github.com/penguinwu/pt2-skill-discovery/tree/main/skills)
- `discovery/{runner,run_case,validate_runner,perf,...}.py` → [`scripts/...`](https://github.com/penguinwu/pt2-skill-discovery/tree/main/scripts)
- `discovery/{design,EXPERIMENT_LIFECYCLE,parallel-runner-design}.md` → [`design/...`](https://github.com/penguinwu/pt2-skill-discovery/tree/main/design)

This issue stays closed on this repo as historical record. New skill-discovery work goes to the new repo's issue tracker (which shares project board #1 with this repo).

**Comment by penguinwu** [comment]
**🔬 Mirror filed:** [`penguinwu/pt2-skill-discovery#5`](https://github.com/penguinwu/pt2-skill-discovery/issues/5) — closed there as well, with paths rewritten to the new repo layout. That's the canonical historical record going forward.
